internal/validation/v2alpha1/hosts.go

```diff
@@ -36,6 +63,33 @@ func validateHosts(parentAttributePath string, vsList networkingv1beta1.VirtualS
 	return failures
 }
 
+func hasSingleHostDefinitionWithCorrectPrefix(gateway *networkingv1beta1.Gateway) bool {
+	host := ""
+	for _, server := range gateway.Spec.Servers {
+		if len(server.Hosts) > 1 {
+			return false
+		}
+		if !strings.HasPrefix(server.Hosts[0], "*.") {
+			return false
+		}
+		if host == "" {
+			host = server.Hosts[0]
+		} else if host != server.Hosts[0] {
+			return false
+		}
+	}
+	return true
+}
+
+func findGateway(gatewayName string, gwList networkingv1beta1.GatewayList) *networkingv1beta1.Gateway {
+	for _, gateway := range gwList.Items {
+		if gatewayName == strings.Join([]string{gateway.Namespace, gateway.Name}, "/") {
```

Comment:
```suggestion
		if gatewayNamespacedName == strings.Join([]string{gateway.Namespace, gateway.Name}, "/") {
```
gatewayName is confusing